pinot-query-planner/src/main/java/org/apache/pinot/query/planner/plannode/MailboxSendNode.java

```diff
@@ -28,7 +30,7 @@
 
 
 public class MailboxSendNode extends BasePlanNode {
-  private final int _receiverStageId;
+  private final BitSet _receiverStages;
```

Comment:
How are you planning to handle the corresponding plan serde changes in a compatible manner?

Reply:
> Originally I though about just changing the current field to add repeat, but it looks like that is problematic on numbers (see https://protobuf.dev/programming-guides/proto3/#updating).

Ah, that's unfortunate, that's exactly what I was curious about.

> Case 1. is impossible as long as we update servers before brokers (which AFAIK is the recommendation)

The recommended order is controller -> broker -> server (https://docs.pinot.apache.org/operators/operating-pinot/upgrading-pinot-cluster).

Reply:
I didn't know that! It sounds a bit strange to me, but anyway, as long as we do not activate spools by default during the migration, the result should be fine anyway